src/background.js

```diff
@@ -1,24 +1,66 @@
-let color = '#3aa757';
-
 chrome.runtime.onInstalled.addListener(() => {
-  chrome.storage.sync.set({ color });
-  console.log('Default background color set to %cbanana', `color: ${color}`);
-  setUp();
+  Crawler.setUp();
 });
 
 chrome.runtime.onStartup.addListener(() => {
-  setUp();
+  Crawler.setUp();
 });
 
 
-function setUp() {
-    console.log("Starting up crawler extension");
-//    chrome.alarms.onAlarm.addListener(runCrawlIteration);
-//    chrome.alarms.create("crawlAlarm", {delayInMinutes: 1.0/60, periodInMinutes: 0.1});
-    setInterval(runCrawlIteration, 1000);
-}
+const Crawler = (() => {
+  var curatedDomains = [];
+
+  return {
+    loadCuratedDomains: () => {
+      const url = chrome.runtime.getURL('../../assets/data/hn-top-domains.json');
+      fetch(url).then(response => {
+        response.json().then(data => {
+          curatedDomains = Object.keys(data);
+          console.log("Loaded curated domains", curatedDomains);
+        });
+      });
+    },
+
+    chooseDomain: () => {
+      time = new Date().getTime();
+      return curatedDomains[time % curatedDomains.length];
```

Comment:
Alternatively, you can use the Math API for this kind of stuff, it'll be more efficient than using a Date object.

Reply:
I could use Math.random(), but that would be seeded by the time anyway, and the JS random function is not very random. I haven't seen another way to get the time since epoch, but if there is one, let me know!